### PR TITLE
Ahad state branch

### PR DESCRIPTION
**Pull request by ABashir01**
**Changes:**

- Added clickable map for election breakdown page that creates state view
- Created map that colors relevant counties based on fake data (please click Alabama for example of it working, this is due to a quirk of how highcharts seems to work - clickability is only granted based on the data and not mapData).
- Set up click for state for future implementation

### CODE REVIEW

File: election-portal/app/election-breakdown/modules/EBMap.tsx [hunk 1]
```diff
@@ -0,0 +1,747 @@
+import React, { useEffect, useState } from "react";
+import { RaceType } from "@/types/RaceType";
+import { Year } from "@/types/Year";
+import Highcharts from "highcharts";
+import HighchartsMap from "highcharts/modules/map";
+import highchartsAccessibility from "highcharts/modules/accessibility";
+import "./EBMap.css";
+
+const presData: FakeData[] = [
```
Comment (arielshif) [inline election-portal/app/election-breakdown/modules/EBMap.tsx:9]: this fake data is out of date. this repo should have more recent fake data

File: election-portal/app/election-breakdown/modules/stateMap.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+}
+
+interface FakeData {
+    "GEOID": string;
+    "value": number;
+}
+
+const colorAxisStops: [number, string][] = [
+  [0, "#B83C2B"], // Republican red
+  [0.25, "#B83C2B"],
```
Comment (arielshif) [inline election-portal/app/election-breakdown/modules/stateMap.tsx:101]: @ashlab11 do you want color gradations or just red/blue

File: election-portal/app/election-breakdown/modules/stateMap.tsx [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    
+    useEffect(() => {
+      retrieveMapData();
+    }, [raceType, year]);
+
+    const retrieveMapData = async() => {
+      const newMapData = await fetchStateGeoJSON(stateName, String(year))
+      setCurrMapData(newMapData);
+      console.log("Set map data in state");
+      console.log(newMapData);
```
Comment (arielshif) [inline election-portal/app/election-breakdown/modules/stateMap.tsx:120]: flagging for consideration

File: election-portal/app/election-breakdown/modules/stateMap.tsx [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+          mapData: mapData,
+          data: presData,
+          joinBy: 'GEOID',
+          nullColor: "#FFFFFF",
+          name: "Predicted Margin",
+          borderColor: "black",
+          events: {
+            click: function(event: any) {
+              const countyName = event.point.properties.NAME;
+              alert(`You just clicked ${countyName}`);
```
Comment (arielshif) [inline election-portal/app/election-breakdown/modules/stateMap.tsx:200]: I assume that this is not permanent but I'm flagging

**Review by arielshif — COMMENTED** [review]
code looks good but I haven't run it

**Review by akslaym — COMMENTED** [review]
General notes needing fix:
- make sure back button takes us back to the national map and not to the homepage here
- Anyway to standardize state sizes? They are all quite differently sized when clicked on which could be a pain to deal with down the line. For example Alaska looks really weird and is in the corner, but Nebraska is HUGE

**Comment by l-ouis** [comment]
> make sure back button takes us back to the national map and not to the homepage here
should be fixed after merging into main
> Anyway to standardize state sizes? They are all quite differently sized when clicked on which could be a pain to deal with down the line. For example Alaska looks really weird and is in the corner, but Nebraska is HUGE
alaska has some weird territory all the way to the right, not sure if that's an error w/ the geoJSONS

**Comment by l-ouis** [comment]
Should be good for another review -- have just been working on this branch to not cause any confusion w/ conflicts

**Comment by arielshif** [comment]
just tested functionality--looks awesome. amazing progress

**Comment by akslaym** [comment]
I am very happy with this personally. I think its good to merge @arielshif

**Comment by arielshif** [comment]
I say merge and add the bug to the backlog

**Review by akslaym — APPROVED** [review]

**Pull request merged**